components/raceDetails.js

```diff
@@ -1,17 +1,34 @@
 import React from 'react';
-import { FlatList, StyleSheet, View } from 'react-native';
-import { Header, List, ListItem, Text } from 'react-native-elements';
+import { View, SectionList } from 'react-native';
+import { List, ListItem, Text } from 'react-native-elements';
 
 export default class RaceList extends React.Component {
     state = {
         race: {},
         refreshing: false
-    }
+    };
+
+    categories = ['zene', 'u16', '16-20', '20-30', '30-40', '40-50', '50+'];
 
     componentDidMount() {
         this.loadRace()
     }
 
+    getSectionsWithData = () => {
+        const sections = [];
+        if(this.state.race.race_results) {
+            this.categories.forEach(category => {
+                sections.push({
+                    title: category.toUpperCase(),
+                    data: this.state.race.race_results.filter(rr => {
```

Comment:
Destructuring would make this expression quite shorter as well.

Maybe `result` instead of `rr` would be easier to read?

Reply:
No need to wrap return statement in `{ }`.
You could also destructure `racer` from `rr`, with simple return as `racer.category === category`